src/features/common/Layout/ProjectPropsContext.tsx

```diff
@@ -1,124 +1,253 @@
+  infoRef: {
+    current: null
+  },
+  exploreContainerRef: null,
   exploreExpanded: false,
-  setExploreExpanded: (value: boolean) => {},
+  setExploreExpanded: () => {},
   exploreForests: false,
-  setExploreForests: (value: boolean) => {},
+  setExploreForests: () => {},
   explorePotential: false,
-  setExplorePotential: (value: boolean) => {},
+  setExplorePotential: () => {},
   exploreDeforestation: false,
-  setExploreDeforestation: (value: boolean) => {},
+  setExploreDeforestation: () => {},
   explorePlanted: false,
-  setExplorePlanted: (value: boolean) => {},
+  setExplorePlanted: () => {},
   infoExpanded: null || '',
-  setInfoExpanded: (value: string) => {},
+  setInfoExpanded: () => {},
   openModal: false,
-  setModalOpen: (value: boolean) => {},
-  viewport: {},
-  setViewPort: (value: {}) => {},
-  setExploreProjects: (value: boolean) => {},
-  mapState: {},
-  setMapState: (value: {}) => {},
+  setModalOpen: () => {},
+  viewport: null,
+  setViewPort: () => {},
+  mapState: null,
+  setMapState: () => {},
   exploreProjects: true,
+  setExploreProjects: () => {},
   loaded: false,
-  setLoaded: (value: boolean) => {},
+  setLoaded: () => {},
   mapRef: {},
   defaultMapCenter: [],
   defaultZoom: 1.4,
   layersSettings: {},
-  setLayersSettings: (value: {}) => {},
+  setLayersSettings: () => {},
   selectedMode: 'location',
-  setSelectedMode: (value: string) => {},
+  setSelectedMode: () => {},
   rasterData: {
     evi: '',
-    imagery: {},
+    imagery: {
+      planetLabs: [],
+      sentinel: [],
+      landsat: [],
+      esri: []
+    },
   },
-  setRasterData: (value: {}) => {},
-  plantLocations: [],
-  setPlantLocations: (value: []) => {},
-  selectedPl: {},
-  setSelectedPl: (value: {}) => {},
+  setRasterData: () => {},
+  selectedPl: null,
+  setSelectedPl: () => {},
   zoomLevel: 1,
-  setZoomLevel: (value: number) => {},
+  setZoomLevel: () => {},
   satellite: false,
-  setSatellite: (value: boolean) => {},
+  setSatellite: () => {},
   plIds: null || [],
-  setPlIds: (value: []) => {},
-  hoveredPl: null || '',
-  setHoveredPl: (value: {}) => {},
+  setPlIds: () => {},
+  hoveredPl: null,
+  setHoveredPl: () => {},
   isPolygonMenuOpen: false,
-  setIsPolygonMenuOpen: (value: boolean) => {},
-  siteViewPort: null || {},
-  setSiteViewPort: (value: {}) => {},
+  setIsPolygonMenuOpen: () => {},
+  siteViewPort: null,
+  setSiteViewPort: () => {},
   filteredProjects: null || [],
-  setFilteredProjects: (value: []) => {},
+  setFilteredProjects: () => {},
   filtersOpen: false,
-  setFilterOpen: (value: boolean) => {},
+  setFilterOpen: () => {},
   purpose: 'trees',
-  setPurpose: (value: string) => {},
+  setPurpose: () => {},
   plantLocationsLoaded: false,
-  setPlantLocationsLoaded: (value: boolean) => {},
-});
+  setPlantLocationsLoaded: () => {},
+  isMobile: false,
+  setIsMobile: () => {},
+};
+
+export const ProjectPropsContext =
+  React.createContext<ProjectContextInterface>(defaultState);
```

Comment:
It would be better to avoid creating a `defaultState` altogether. We are already setting up defaults using local state within `ProjectPropsProvider`. Simply initialize context as null while creating it.

This will also address the TS error on line 190.

```Argument of type '{ project: {} | null; setProject: () => void; projects: never[]; setProjects: () => void; plantLocations: never[]; setPlantLocations: () => void; showSingleProject: boolean; setShowSingleProject: () => void; ... 66 more ...; setIsMobile: () => void; }' is not assignable to parameter of type 'ProjectContextInterface'.
  Types of property 'project' are incompatible.
    Type '{} | null' is not assignable to type 'Project | null'.
      Type '{}' is missing the following properties from type 'Project': id, _scope, allowDonations, classification, and 44 more.```

Reply:
Note: If you do this, you will need to handle the unlikely scenario that the context is `null`. This can be handled by creating a hook similar to how it is done for `BulkCodeProvider`.

```export const useBulkCode = (): BulkCodeContextInterface => {
  const context = useContext(BulkCodeContext);
  if (!context) {
    throw new Error('BulkCodeContext must be used within BulkCodeProvider');
  }
  return context;
};```

Reply:
@sunilsabatp can you take over this pull request again please.